### DISCUSSION TITLE
Ideas to help with keyboard

### DISCUSSION THREAD

**Discussion by UkuleleVillain** [OnionUI/Onion#1265] [Ideas]
[NVM, i was talking about one of miyoo's built in keyboards]
I'm not a professional or anything, but these are some of my thoughts on the onion keyboard and some of the changes that might be helpful to the user experience. 

1. The delete button should start deleting repeatedly if held down for a bit, just like most online keyboards do.

2. A mouse cursor or a way to navigate the text box might be helpful, but this could be a non-issue if the below problem is solved.

3. the current system has a delay on when you press the A button and when the key is typed. this leads to things where i'm hovering over the a key, press A to enter that key, then press right to move, and it inputs S instead. 

This seem illogical, because even if there was a delay, there would have to be a different delay for the d pad and ABXY buttons for the right to be registered before the A press. 

Even if it is just a hardware issue, it can easily be solved by checking what the selector was hovering over a bit before the A input is received. this would counter the delay and solve the issue.

4. The switch between letters and symbols should show the opposites of what is currently being read, not the same. This removes confusion when trying to find how to type a symbol.

5. 1 click of the caps button should do one capital letter before returning to lowercase, two clicks should enable caps lock.

These fixes just make it more in line with the way that most modern keyboards work, and create a smoother experience.

**Comment by Aemiii91** [comment]
Can you please provide a screenshot of the keyboard you're referring to? (The Onion keyboard is the one used in Search, whereas the MainUI keyboard, made by Miyoo, is used for WiFi - not for much longer though)

**Reply by UkuleleVillain** [reply]
Wow, sorry i thought that the wifi keyboard was part of onion os. the game searching keyboard is perfect!
